android/src/main/java/com/bitmovin/player/reactnative/PlayerModule.kt

```diff
@@ -83,360 +82,339 @@ class PlayerModule(private val context: ReactApplicationContext) : ReactContextB
         }
     }
 
     /**
-     * Call `.pause()` on `nativeId`'s player.
+     * Call `.pause()` on [nativeId]'s player.
      * @param nativeId Target player Id.
      */
     @ReactMethod
-    fun pause(nativeId: NativeId) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.pause()
+    fun pause(nativeId: NativeId, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).pause()
         }
     }
 
     /**
-     * Call `.seek(time:)` on `nativeId`'s player.
+     * Call `.seek(time:)` on [nativeId]'s player.
      * @param nativeId Target player Id.
      * @param time Seek time in seconds.
      */
     @ReactMethod
-    fun seek(nativeId: NativeId, time: Double) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.seek(time)
+    fun seek(nativeId: NativeId, time: Double, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).seek(time)
         }
     }
 
     /**
-     * Call `.timeShift(offset:)` on `nativeId`'s player.
+     * Call `.timeShift(offset:)` on [nativeId]'s player.
      * @param nativeId Target player Id.
      * @param offset Offset time in seconds.
      */
     @ReactMethod
-    fun timeShift(nativeId: NativeId, offset: Double) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.timeShift(offset)
+    fun timeShift(nativeId: NativeId, offset: Double, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).timeShift(offset)
         }
     }
 
     /**
-     * Call `.mute()` on `nativeId`'s player.
+     * Call `.mute()` on [nativeId]'s player.
      * @param nativeId Target player Id.
      */
     @ReactMethod
-    fun mute(nativeId: NativeId) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.mute()
+    fun mute(nativeId: NativeId, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).mute()
         }
     }
 
     /**
-     * Call `.unmute()` on `nativeId`'s player.
+     * Call `.unmute()` on [nativeId]'s player.
      * @param nativeId Target player Id.
      */
     @ReactMethod
-    fun unmute(nativeId: NativeId) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.unmute()
+    fun unmute(nativeId: NativeId, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).unmute()
         }
     }
 
     /**
-     * Call `.destroy()` on `nativeId`'s player.
+     * Call `.destroy()` on [nativeId]'s player.
      * @param nativeId Target player Id.
      */
     @ReactMethod
-    fun destroy(nativeId: NativeId) {
-        uiManager()?.addUIBlock {
-            players[nativeId]?.let {
-                it.destroy()
-                players.remove(nativeId)
-            }
+    fun destroy(nativeId: NativeId, promise: Promise) {
+        promise.resolveOnUIThread {
+            getPlayer(nativeId).destroy()
+            players.remove(nativeId)
```

Comment:
This block will resolve with the removed player instance. That's not what we want.
Maybe introducing a function that resolves without a return value could reduce such accidental returns?

Reply:
This is just one place I spotted, but there might be more where we accidentally return something.

Reply:
With the new type safe promise, this should not occur.